lib/src/screens/supa_phone_auth.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:flutter/material.dart';
+import 'package:supabase_auth_ui/src/utils/constants.dart';
+import 'package:supabase_auth_ui/src/utils/supabase_auth_ui.dart';
+import 'package:supabase_flutter/supabase_flutter.dart';
+
+class SupaPhoneAuth extends StatefulWidget {
+  final String redirectUrl;
+
+  const SupaPhoneAuth({Key? key, required this.redirectUrl}) : super(key: key);
+
+  @override
+  State<SupaPhoneAuth> createState() => _SupaPhoneAuthState();
+}
+
+class _SupaPhoneAuthState extends State<SupaPhoneAuth> {
+  final _formKey = GlobalKey<FormState>();
+  final _phone = TextEditingController();
+  final _password = TextEditingController();
+
+  SupabaseAuthUi supaAuth = SupabaseAuthUi();
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _phone.dispose();
+    _password.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Form(
+      autovalidateMode: AutovalidateMode.onUserInteraction,
+      key: _formKey,
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          TextFormField(
+            validator: (value) {
+              if (value == null || value.isEmpty) {
+                return 'Please enter a valid email address';
+              }
+              return null;
+            },
+            decoration: const InputDecoration(
+              prefixIcon: Icon(Icons.phone),
+              hintText: 'Enter your phone number',
+            ),
+            controller: _phone,
+          ),
+          spacer(16),
+          TextFormField(
+            validator: (value) {
+              if (value == null || value.isEmpty || value.length < 6) {
+                return 'Please enter a password that is atleast 6 characters long';
+              }
+              return null;
+            },
+            decoration: const InputDecoration(
+              prefixIcon: Icon(Icons.lock),
+              hintText: 'Enter your password',
+            ),
+            obscureText: true,
+            controller: _password,
+          ),
+          spacer(16),
+          ElevatedButton(
+            child: const Text(
+              'Sign Up',
+              style: TextStyle(fontWeight: FontWeight.bold),
+            ),
+            onPressed: () async {
+              if (!_formKey.currentState!.validate()) {
+                return;
+              }
+
+              try {
+                await supaAuth.signInUserWithPhone(_phone.text, _password.text);
+                successAlert;
+                if (!mounted) return;
+                Navigator.popAndPushNamed(context, widget.redirectUrl, arguments: {"phone": _phone.text});
```

Comment:
Should we call the successAlert method here?
```suggestion
                if (!mounted) return;
                await successAlert(context);
                if(mouted) {
                  Navigator.popAndPushNamed(context, widget.redirectUrl, arguments: {"phone": _phone.text});
                }
```

Reply:
I was thinking tell them it was successful before redirecting them to where they enter the code sent to the number. What were your thoughts?
Also, do we need to explicitly check for mounted?

Reply:
I agree with you! 

This doesn't show the dialog though, right?
```dart
successAlert;
```

>Also, do we need to explicitly check for mounted?
According to the new lint rule we do, but we can ignore them if you want to.